OpenTracing/OpenTracing.OpenTracing/ITracer.cs

```diff
@@ -0,0 +1,13 @@
+﻿using OpenTracing.Propagation;
+
+namespace OpenTracing
+{
+    public interface ITracer<T>
+    {
+        ISpan<T> StartSpan(string operationName);
+        ISpan<T> StartSpan(StartSpanOptions startSpanOptions);
+
+        void Inject(ISpan<T> span, IInjectCarrier<T> carrier);
+        bool TryJoin(string operationName, IExtractCarrier<T> carrier, out ISpan<T> span);
```

Comment:
I'm surprised about the `out ISpan<T> span`: why not have `TryJoin` return that type instead? It also seems like there will be public/private problems trying to populate `out`, though maybe I just don't know C# well enough...